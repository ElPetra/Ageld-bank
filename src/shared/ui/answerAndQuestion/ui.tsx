import React, { useState } from 'react';

import { Icon } from '../icon';
import './style.scss';

interface Props {
    question: string;
    answer: string;
}

export const AnswerAndQuestion: React.FC<Props> = ({ question, answer }) => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleAnswer = () => {
        setIsOpen(prevState => !prevState);
    };

    return (
        <div className='block'>
            <div className='question'>
                <p className='question__text'>{question}</p>
                <button className='question__button' onClick={toggleAnswer}>
                    <Icon
                        width={24}
                        icon={isOpen ? 'arrow-up' : 'arrow-down'}
                    />
                </button>
            </div>
            {isOpen && (
                <div className='answer'>
                    <p className='answer__text'>{answer}</p>
                </div>
            )}
        </div>
    );
};
