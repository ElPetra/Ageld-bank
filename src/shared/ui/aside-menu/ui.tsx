import cn from 'classnames';

import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    visible: boolean;
    children: ReactNode;
}

export const AsideMenu = ({ visible, children }: Props) => {
    const asideClass = cn('aside-menu', {
        visible: visible
    });
    return (
        <>
            {visible && <div className='overlay' />}
            <aside className={asideClass}>
                <div className='aside-menu__content'>{children}</div>
            </aside>
        </>
    );
};
