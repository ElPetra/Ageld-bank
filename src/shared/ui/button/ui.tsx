import { memo } from 'react';

import cn from 'classnames';

import type { ButtonHTMLAttributes, ReactNode } from 'react';

import './styles.scss';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
    type?: 'submit' | 'reset' | 'button';
    disabled?: boolean;
    size?: 'medium' | 'large';
    variant?: 'primary' | 'secondary' | 'link';
    width?: 'auto' | 'max';
    status?: 'active' | undefined;
    children: ReactNode;
}

export const Button = memo(
    ({
        type,
        size = 'medium',
        disabled = false,
        variant = 'primary',
        width = 'auto',
        status,
        children,
        ...props
    }: Props) => {
        const buttonClass = cn(size, width, status, {
            button: variant !== 'link',
            [variant]: !disabled
        });
        return (
            <button
                type={type || 'button'}
                disabled={disabled}
                className={buttonClass}
                {...props}
            >
                {children}
            </button>
        );
    }
);
