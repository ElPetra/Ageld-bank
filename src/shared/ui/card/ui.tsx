import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    children: ReactNode;
    gap?: 'small' | 'medium' | 'large';
    direction?: 'column' | 'row';
    status?: 'active' | '';
}

export const Card = ({
    children,
    gap = 'medium',
    direction = 'column',
    status
}: Props) => (
    <div className={`card ${gap + '-gap'} ${direction} ${status}`}>
        {children}
    </div>
);
