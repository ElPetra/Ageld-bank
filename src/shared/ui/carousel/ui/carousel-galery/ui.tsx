import cn from 'classnames';

import type { RefObject } from 'react';
import './styles.scss';

interface Props {
    containerRef: RefObject<HTMLDivElement>;
    distance: number;
    angle: number;
    cards: Array<JSX.Element>;
}

export const Gallery = ({ containerRef, distance, angle, cards }: Props) => {
    const innerContainerClasses = cn({
        carousel_inner_container: cards.length === 3,
        carousel_inner_container_hidden: cards.length > 3
    });
    return (
        <div className={innerContainerClasses}>
            <div className='carousel_container' ref={containerRef}>
                <div
                    className='carousel_element'
                    style={{ transform: `translateZ(${distance}px)` }}
                >
                    {cards[0]}
                </div>
                {distance &&
                    cards.map((el, i) => {
                        if (i !== 0) {
                            return (
                                <div
                                    key={i}
                                    className='carousel_element'
                                    style={{
                                        transform: `rotateY(${i * angle}deg) translateZ(${distance}px)`
                                    }}
                                >
                                    {el}
                                </div>
                            );
                        }
                        return null;
                    })}
            </div>
        </div>
    );
};
