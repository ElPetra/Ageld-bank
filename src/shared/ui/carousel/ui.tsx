import { useRef } from 'react';
import { useCarousel, useCarouselControls } from 'src/shared/lib/carousel';

import { CarouselInner } from './ui/inner';
import { CarouselIndicators } from './ui/indicators';
import { CarouselControls } from './ui/controls';

import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    cards: ReactNode[];
}

export const Carousel = ({ cards }: Props) => {
    const containerRef = useRef<HTMLDivElement>(null);
    const { angle, distance } = useCarousel(cards.length);
    const { changeRotateByIndicator, changeRotateByControl, active } =
        useCarouselControls(cards.length, containerRef, angle);
    return (
        <div className='carousel'>
            <div>
                <CarouselInner
                    distance={distance}
                    angle={angle}
                    active={active}
                    containerRef={containerRef}
                    cards={cards}
                />
                <CarouselIndicators
                    cards={cards}
                    changeRotateByIndicator={changeRotateByIndicator}
                    active={active}
                />
            </div>
            <CarouselControls changeRotateByControl={changeRotateByControl} />
        </div>
    );
};
