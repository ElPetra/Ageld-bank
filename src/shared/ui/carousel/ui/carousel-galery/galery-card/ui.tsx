import { Button, Card, Text } from 'src/shared/ui';
import './styles.scss';
import { type ReactNode } from 'react';

interface Props {
    title: string;
    content: string | ReactNode;
    button: string;
}

export const GalleryCard = ({ title = 'Новинка!', content, button }: Props) => {
    return (
        <Card
            className='gallery_card'
            justify='space-between'
            direction='column'
            gap='small'
            borderRadius='large'
            padding='medium'
        >
            <Text size='m' weight='bold' align='center'>
                {title}
            </Text>
            <Text size='m'>{content}</Text>
            <Button variant='secondary' size='medium' width='max'>
                {button}
            </Button>
        </Card>
    );
};
