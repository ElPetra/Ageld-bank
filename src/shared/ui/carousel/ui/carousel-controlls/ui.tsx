import cn from 'classnames';
import { type RefObject } from 'react';
import { useCarouselControlls } from 'src/shared/lib/carousel';
import './styles.scss';
import { useTranslation } from 'react-i18next';

interface Props {
    containerRef: RefObject<HTMLDivElement>;
    angle: number;
}

export const Controlls = ({ containerRef, angle }: Props) => {
    const { changeRotateByArrow, changeRotateByControl, active } =
        useCarouselControlls(containerRef, angle);
    const { t } = useTranslation();
    const controllClasses = (el: 'center' | 'prev' | 'next') =>
        cn({
            controll: true,
            active_controll: el === active
        });
    return (
        <div className='carousel_controlls_container'>
            <div className='controll_block_container'>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('prev')}
                >
                    <div
                        className={controllClasses('prev')}
                        aria-label={t('предыдущий слайд')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('center')}
                >
                    <div
                        className={controllClasses('center')}
                        aria-label={t('актуальный слайд')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('next')}
                >
                    <div
                        className={controllClasses('next')}
                        aria-label={t('следующий слайд')}
                        role='button'
                    ></div>
                </div>
            </div>
            <div className='controll_block_container arrows'>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByArrow('left')}
                >
                    <div
                        className='arrow controll'
                        aria-label={t('переключить назад')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block right_arrow'
                    onClick={() => changeRotateByArrow('right')}
                >
                    <div
                        className='arrow controll'
                        aria-label={t('переключить вперед')}
                        role='button'
                    ></div>
                </div>
            </div>
        </div>
    );
};
