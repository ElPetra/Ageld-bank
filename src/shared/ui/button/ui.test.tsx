import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/jest-globals';

import { Button } from './ui';

describe('Button ui', () => {
    test('match snapshot', () => {
        render(<Button>Button</Button>);
        const button = screen.getByText('Button');
        expect(button).toMatchSnapshot();
    });

    test('link match snapshot', () => {
        render(<Button variant='link'>Link</Button>);
        const link = screen.getByText('Link');
        expect(link).toMatchSnapshot();
    });

    test('button renders', () => {
        render(<Button>Button</Button>);
        const button = screen.getByText('Button');
        expect(button).toBeInTheDocument();
    });

    test('onClick have been called', () => {
        const onClickMock = jest.fn();
        render(<Button onClick={onClickMock}>Button</Button>);
        const button = screen.getByText('Button');
        expect(button).toBeInTheDocument();
        fireEvent.click(button);
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    test('onClick have not been called', () => {
        const onClickMock = jest.fn();
        render(
            <Button onClick={onClickMock} disabled>
                Button
            </Button>
        );
        const button = screen.getByText('Button');
        expect(button).toBeInTheDocument();
        fireEvent.click(button);
        expect(onClickMock).toHaveBeenCalledTimes(0);
    });
});
