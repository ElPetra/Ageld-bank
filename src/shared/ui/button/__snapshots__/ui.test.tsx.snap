// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ui link match snapshot 1`] = `
<button
  class="medium auto link"
  type="button"
>
  Link
</button>
`;

exports[`Button ui match snapshot 1`] = `
<button
  class="medium auto button primary"
  type="button"
>
  Button
</button>
`;
